ui-common/src/components/UI/KeyValueTable/KeyValueTable.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { useEffect, useState } from "react";
+import styles from "./KeyValueTable.module.sass"
+import deleteIcon from "delete.svg"
+import deleteIconActive from "delete-active.svg"
+import HoverImage from "../HoverImage/HoverImage";
+
+interface KeyValueTableProps {
+    data: any
+    onDataChange: (data: any) => void
+    keyPlaceholder?: string
+    valuePlaceholder?: string
+}
+
+type Row = { key: string, value: string }
+
+const KeyValueTable: React.FC<KeyValueTableProps> = ({ data, onDataChange, keyPlaceholder, valuePlaceholder }) => {
+
+    const [keyValueData, setKeyValueData] = useState([] as Row[])
+
+    useEffect(() => {
+        if (!data) return;
+        const currentState = [...data, { key: "", value: "" }]
+        setKeyValueData(currentState)
+    }, [data])
```

Comment:
I saw you resolved the last previous comment but didn't add this "setKeyValueData" to the deps array.
why does it not cause an eslint warning?

Reply:
i dont know really